packages/core/src/types.ts

```diff
@@ -207,3 +208,48 @@ export type ClientStorageType =
   | 'react-native'
   | 'web'
   | 'cookie'
+
+// Hasura-auth API response types
+interface NullableErrorResponse {
+  error: ErrorPayload | null
+}
+
+export type NhostSessionResponse =
+  | { session: null; error: ErrorPayload }
+  | { session: NhostSession | null; error: null }
+
+export type SignUpResponse = NhostSessionResponse
+
+export interface SignInResponse {
```

Comment:
It seems that this type extends `NhostSessionResponse`.

Reply:
What do you mean? I don't understand

Reply:
This is irrelevant, since we are not changing `NhostSessionResponse. (see my comment here: https://github.com/nhost/nhost/pull/907#discussion_r958093496)